src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -103,7 +103,7 @@ public TestResult(ITest test)
             Test = test;
             ResultState = ResultState.Inconclusive;
 
-#if NETSTANDARD1_4
+#if !PARALLEL
```

Comment:
The idea was to enable locking around the OutWriter by the same switch that enables the RwLock around the other properties. I should probably have switched the If around instead of using `!PARALLEL`.

Reply:
The PARALLEL define means "build in parallel support" so it's related to platforms but isn't really the same thing. For example, we used to have a compact framework build. At one time we didn't support parallel on it but later we did. So this is a policy switch that defines NUnit capabilities, not specifically a platform switch.

In the case of Synchronized, we do it on all platforms that have it. .NET 1.4 doesn't have that property. As you have modified it, any platform that doesn't build in parallel execution will try to use Synchronized. Are all platforms building? If so, maybe the conditional compile is obsolete. I'm not up on what we are and aren't building these days, so someone else will need to check.

Bottom line, however, we probably __do__ want the lock on all builds that define PARALLEL. It's just the Synchronized bit that seems in question.

Thing to check is whether all writes will actually use the lock. Depends on whether they write to Output or _output.

Reply:
OK. That makes sense, at least until / unless we want to support parallel tests on a platform without a synchronized text writer. That seems unlikely.

Reply:
Thinking about this again: As I'm not 100% sure about where exactly the threading issue comes from I'm thinking we might have to back port the synchronized TextWriter back to NetStandard1.4 - or is it guaranteed that there is no concurrent access to OutWriter on that platform (reading or writing).